js/engine/src/index.d.ts

```diff
@@ -12,7 +12,8 @@ export interface Engine {
   execute: (
     language: string,
     rulesFilename: string,
-    targetFilename: string
+    root: string,
+    targetFilenames: string[]
```

Comment:
re: my question about semgrep-app, can we make this `string[] | string` and coerce `string` to `string[]` inside index.js?

Reply:
execute right now is a direct ref to the ocaml jsoo output. We can make execute an anonymous function that calls it, and coerce, but I'm not sure I see a reason to. With this PR oss semgrep js can scan multiple files, and I don't see a reason to remove that, even if we're not using it right now. Also we're the only ones using this interface, so when we update semgrep-app (see response above), I'll just fix it and things should be ok. I'm going to have to do this to support the root field anyways, which we need if we want semgrep pro js, and want to keep the interfaces/build process simple.